src/main/java/mate/academy/springboot/swagger/controller/ProductController.java

```diff
@@ -0,0 +1,104 @@
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final ProductMapper productMapper;
+    private final SortUtil sortUtil;
+
+    @GetMapping
+    @Operation(summary = "Find all products")
+    public List<ProductResponseDto> findAll(@RequestParam (defaultValue = "10")
+                                                @Parameter(description =
+                                                        "Numbers of product on 1 page")
+                                                        Integer count,
+                                            @RequestParam (defaultValue = "0")
+                                                    @Parameter(description = "Number of page")
+                                                    Integer page,
+                                            @RequestParam (defaultValue = "id")
+                                                @Parameter(description = "Parameter for sorting")
+                                                        String sortBy) {
+        Sort sort = Sort.by(sortUtil.getSort(sortBy));
+        PageRequest pageRequest = PageRequest.of(page, count, sort);
+        return productService.findAll(pageRequest).stream()
+                .map(productMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/{id}")
+    @Operation(summary = "Get product by id")
+    public ProductResponseDto getById(@PathVariable Long id) {
+        return productMapper.toDto(productService.findById(id));
+    }
+
+    @DeleteMapping("/{id}")
+    @Operation(summary = "Delete product by id")
+    public void delete(@PathVariable Long id) {
+        productService.delete(id);
+    }
+
+    @PutMapping("/{id}")
+    @Operation(summary = "Update product by id")
+    public ProductResponseDto update(@PathVariable Long id,
+                                     @Schema(description = "Update product",
+                                     required = true, implementation = ProductRequestDto.class)
+                                     @RequestBody @Valid ProductRequestDto requestDto) {
+        Product product = productMapper.toModel(requestDto);
+        product.setId(id);
+        return productMapper.toDto(productService.update(product));
+    }
+
+    @PostMapping
+    @Operation(summary = "Create product and add to DB")
+    public ProductResponseDto save(@Schema(description = "Create product",
+            required = true, implementation = ProductRequestDto.class)
+            @Valid @RequestBody ProductRequestDto requestDto) {

```

Comment:
same

or if you don't want to remove this annotation, add validating annotations to RequestDto